File: Algorithms/p038_Count_and_Say/p038_Count_and_Say.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        s = '1'
        n -= 1
        while n>0:
            s1 = []
            pos = 0
            while pos<len(s):
                start = pos
                while pos<len(s) and s[pos]==s[start]:
                    pos += 1
                s1.append(str(pos-start))
                s1.append(s[start])
            s = ''.join(s1)
            n -= 1
        return s

File: Algorithms/p038_Count_and_Say/p038_Count_and_Say.cpp
class Solution {
public:
    string countAndSay(int n) {
        string s("1");
        while(--n>0){
            string s1;
            int pos = 0;
            while(pos<s.size()){
                int start = pos;
                while(pos<s.size()&&s[pos]==s[start]) ++pos;
                s1 += to_string(pos-start)+s[start];
            }
            s = move(s1);
        }
        return s;
    }
};
